do not omit zero TTL values in answer section (fix #165)

stklcode opened the pull request:
TTL is a mandatory value of the RR header and zero is a valid value.
However the Go JSON marshaller cannot distinguish between unset and zero numeric values so the result got dropped.

From what I think valid records with missing TTL (not zero, but completely missing) would be completely omitted in previous steps so the simple solution of not omitting "empty" (i.e. zero) values in the output should be OK.

stklcode committed:
do not omit zero TTL values in answer section (fix #165)

TTL is a mandatory value of the RR header and zero is a valid value.
However the Go JSON marshaller cannot distinguish between unset and zero
numeric values so the result got dropped.

zakird reviewed (approved):
agree. thanks for fixing.

zakird commented:
Can you fix the corresponding test? Once that's done, I'm happy to merge.

stklcode commented:
It not a "corresponding" test that fails. Apparently Google has changed the PTR of 8.8.8.8 from "google-public-dns-a.google.com" to "dns.google" recently...

Fixed it anyway. (probably should not be squashed during merge as the changes are not related to the actual issue)

stklcode committed:
test: update expected PTR for 8.8.8.8

Google recently changed the record to "dns.google" so the expected
result needs to be adjusted for integration tests.

zakird commented:
Thanks for taking care of that.
